tfsec failed to run

tfsec has been working fine for a few weeks, however today it started to display and error stating that "tfsec failed to run"

Here is the tfsec output from vscode:

```zsh
[DEBUG][2022-02-01 22:13:02.060235 +0000 GMT m=+8.901527922] Starting scanner...

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x14dab60]

goroutine 1 [running]:

github.com/zclconf/go-cty/cty.Type.FriendlyName(...)
	/home/runner/work/tfsec/tfsec/vendor/github.com/zclconf/go-cty/cty/type.go:47
github.com/hashicorp/hcl/v2/hclsyntax.(*ConditionalExpr).Value(0xc0003ef1f0, 0xc01f7a0cd8)
	/home/runner/work/tfsec/tfsec/vendor/github.com/hashicorp/hcl/v2/hclsyntax/expression.go:623 +0x1320
github.com/aquasecurity/tfsec/internal/pkg/block.getStrings({0x1911978, 0xc0003ef1f0}, 0x10)
	/home/runner/work/tfsec/tfsec/internal/pkg/block/attribute.go:254 +0x208
github.com/aquasecurity/tfsec/internal/pkg/block.(*Attribute).ValueAsStrings(...)
	/home/runner/work/tfsec/tfsec/internal/pkg/block/attribute.go:243
github.com/aquasecurity/tfsec/internal/pkg/adapter/aws/vpc.adaptSGRule(0xc00035da40, {0xc005380000, 0x230, 0x400})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/aws/vpc/adapt.go:142 +0x28a
github.com/aquasecurity/tfsec/internal/pkg/adapter/aws/vpc.(*sgAdapter).adaptSecurityGroup(0xc0031dd6d8, 0xc00045ea00, {0xc005380000, 0x230, 0x400})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/aws/vpc/adapt.go:110 +0x793
github.com/aquasecurity/tfsec/internal/pkg/adapter/aws/vpc.(*sgAdapter).adaptSecurityGroups(0xc0031dd6d8, {0xc005380000, 0x230
, 0x400})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/aws/vpc/adapt.go:45 +0x11b
github.com/aquasecurity/tfsec/internal/pkg/adapter/aws/vpc.Adapt({0xc005380000, 0x230, 0x400})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/aws/vpc/adapt.go:16 +0x156
github.com/aquasecurity/tfsec/internal/pkg/adapter/aws.Adapt({_, _, _})
	
/home/runner/work/tfsec/tfsec/internal/pkg/adapter/aws/adapt.go:76 +0xeff
github.com/aquasecurity/tfsec/internal/pkg/adapter.Adapt({0xc005380000, 0x230, 0x400})
	/home/runner/work/tfsec/tfsec/internal/pkg/adapter/adapt.go:19 +0x4e
github.com/aquasecurity/tfsec/internal/pkg/scanner.(*Scanner).Scan(0xc006cbc660, {0xc005380000, 0x0
, 0x400})
	/home/runner/work/tfsec/tfsec/internal/pkg/scanner/scanner.go:51 +0x9f
github.com/aquasecurity/tfsec/internal/app/tfsec/cmd.glob..func1(0x1e81320, {0xc0003ee230, 0x1, 0x7
})
	/home/runner/work/tfsec/tfsec/internal/app/tfsec/cmd/root.go:114 +0x7ec
github.com/spf13/cobra.(*Command).execute(0x1e81320, {0xc00001e090, 0x7, 0x7
})
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:856 +0x60e
github.com/spf13/cobra.(*Command).ExecuteC(0x1e81320)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go:974 +0x3bc
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/work/tfsec/tfsec/vendor/github.com/spf13/cobra/command.go
:902
main.main()
	/home/runner/work/tfsec/tfsec/cmd/tfsec/main.go:11 +0x25
```

**Desktop (please complete the following information):**
 - OS: macOS 10.15.7
 - Tfsec Version: v1.0.7
 - VSCode Version: v1.7.4

I was experimenting with setting: "tfsec.ignoreDownloadedModules": to false. It ran OK, but I have changed it back to true.

I have also restarted vscode and reinstalled the extension.

I tried upgrading tfsec to v1.0.8 also but the problem persists.

I should also add, tfsec runs fine from the cli.

--- a/internal/pkg/block/attribute.go
+++ b/internal/pkg/block/attribute.go
@@ -248,6 +248,7 @@ func (attr *Attribute) ValueAsStrings() []string {
 	return getStrings(attr.hclAttribute.Expr, attr.ctx.Inner())
 }
 
+//nolint
 func getStrings(expr hcl.Expression, ctx *hcl.EvalContext) []string {
 
 	defer func() {

--- a/internal/pkg/block/attribute.go
+++ b/internal/pkg/block/attribute.go
@@ -177,6 +177,11 @@ func (attr *Attribute) Each(f func(key cty.Value, val cty.Value)) {
 	if attr == nil {
 		return
 	}
+	defer func() {
+		if err := recover(); err != nil {
+			debug.Log("go-cty bug detected - cannot call ForEachElement: %s", err)
+		}
+	}()
 	val := attr.Value()
 	val.ForEachElement(func(key cty.Value, val cty.Value) (stop bool) {
 		f(key, val)
@@ -244,6 +249,13 @@ func (attr *Attribute) ValueAsStrings() []string {
 }
 
 func getStrings(expr hcl.Expression, ctx *hcl.EvalContext) []string {
+
+	defer func() {
+		if err := recover(); err != nil {
+			debug.Log("go-cty bug detected - failed to derive value from expression: %s", err)
+		}
+	}()
+
 	var results []string
 	switch t := expr.(type) {
 	case *hclsyntax.TupleConsExpr:

